boards/common/esp32x/include/board_common.h

```diff
@@ -103,25 +103,45 @@ extern "C" {
 #define SPI_FLASH_DRIVE_START  0
 #endif
 
-#define MTD_0 mtd0          /**< Flash MTD device */
-extern mtd_dev_t *mtd0;     /**< Flash MTD device pointer */
+/** Default Flash MTD drive definition */
+#ifndef MTD_0
+#define MTD_0 mtd0
+#endif
+
+/** Pointer to the default Flash MTD drive structure */
+extern mtd_dev_t *mtd0;
 
 #if MODULE_MTD_SDCARD_DEFAULT || DOXYGEN
 
-#define MTD_1 mtd1          /**< SD Card MTD device */
-extern mtd_dev_t *mtd1;     /**< SD Card MTD device pointer */
+/** Default SD Card/MMC MTD drive definition */
+#ifndef MTD_1
+#define MTD_1 mtd1
+#endif
+/** Pointer to the default SD Card/MMC MTD drive structure */
+extern mtd_dev_t *mtd1;
```

Comment:
Those aren't needed anymore, right?

Reply:
Ah I thought applications are expected to only use `mtd_dev_get(0)` instead of `MTD_0` now.

Reply:
Maybe we should move this part https://github.com/RIOT-OS/RIOT/blob/a24066be93dd7494ba17c0f38b76ec24aee7ed97/drivers/include/mtd_default.h#L38-L59 to `drivers/mtd.h` and remove all `extern mtd_dev_t *mtd<n>` declarations from `board.h`?

Reply:
`tests/pkg/littlefs` uses for example `MTD_0` to decide whether it can use a hardware MTD.

Reply:
@benpicco What do you think about the approach above?

Reply:
The thing is that the move to XFA was already an API breaking change, so I think it would be fine to also move away from `MTD_x` at the same time.

Or maybe we just 

```C
#define MTD_0 mtd_dev_get(0)
```

so the required changes are contained to the boards.

Reply:
> Or maybe we just
> 
> ```c
> #define MTD_0 mtd_dev_get(0)
> ```
> 
> in `mtd.h` so the required changes are contained to the boards.

That was the initial idea, but then my concern was that we wouldn't be able to check via preprocessor definitions whether a board has MTDs at all.